Storage/container/many_to_many.h

```diff
@@ -238,27 +220,27 @@ class GenericMatrix {
 };
 
 template <typename T, typename UPDATE_EVENT, typename DELETE_EVENT>
-using UnorderedMatrix = GenericMatrix<T, UPDATE_EVENT, DELETE_EVENT, Unordered, Unordered>;
+using UnorderedManyToMany = GenericManyToMany<T, UPDATE_EVENT, DELETE_EVENT, Unordered, Unordered>;
 
 template <typename T, typename UPDATE_EVENT, typename DELETE_EVENT>
-using OrderedMatrix = GenericMatrix<T, UPDATE_EVENT, DELETE_EVENT, Ordered, Ordered>;
+using OrderedManyToMany = GenericManyToMany<T, UPDATE_EVENT, DELETE_EVENT, Ordered, Ordered>;
 
 }  // namespace container
 
 template <typename T, typename E1, typename E2>  // Entry, update event, delete event.
-struct StorageFieldTypeSelector<container::UnorderedMatrix<T, E1, E2>> {
-  static const char* HumanReadableName() { return "UnorderedMatrix"; }
+struct StorageFieldTypeSelector<container::UnorderedManyToMany<T, E1, E2>> {
+  static const char* HumanReadableName() { return "UnorderedManyToMany"; }
 };
 
 template <typename T, typename E1, typename E2>  // Entry, update event, delete event.
-struct StorageFieldTypeSelector<container::OrderedMatrix<T, E1, E2>> {
-  static const char* HumanReadableName() { return "OrderedMatrix"; }
+struct StorageFieldTypeSelector<container::OrderedManyToMany<T, E1, E2>> {
+  static const char* HumanReadableName() { return "OrderedManyToMany"; }
 };
 
 }  // namespace storage
 }  // namespace current
 
-using current::storage::container::UnorderedMatrix;
-using current::storage::container::OrderedMatrix;
+using current::storage::container::UnorderedManyToMany;
+using current::storage::container::OrderedManyToMany;
```

Comment:
Perhaps even not here but into a `matrix.h` file (which should just be one `#include` and two `using`-s for a few commits from now on)?